src/visitors/CMakeLists.txt

```
@@ -66,9 +66,17 @@ add_library(visitor_obj OBJECT ${VISITOR_SOURCES} ${VISITOR_GENERATED_SOURCES})
 set_property(TARGET visitor_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 add_dependencies(visitor_obj lexer_obj)
-
 add_library(visitor STATIC $<TARGET_OBJECTS:visitor_obj>)
-target_link_libraries(visitor PRIVATE pybind11::embed)
+
+# ~~~
+# pybind11::embed adds PYTHON_LIBRARIES to target_link_libraries. To avoid link to
+# libpython, we can use `module` interface library from pybind11.
+# ~~~
+if(NOT LINK_AGAINST_PYTHON)
+  target_link_libraries(visitor PRIVATE module)
```

```suggestion
  target_link_libraries(visitor PRIVATE pybind11::module)
```
https://pybind11.readthedocs.io/en/stable/compiling.html?highlight=cmake#advanced-interface-library-target
unless I'm again misunderstanding/forgetting something...

---

no, it was oversight from me. I didn't read doc but was looking at CMakeLists.txt of pybind11 and didn't notice `ALIAS`. FIXED.